x-pack/plugins/fleet/server/services/epm/elasticsearch/template/template.ts

```diff
@@ -532,20 +540,13 @@ const updateExistingDataStream = async ({
     );
     // if update fails, rollover data stream
   } catch (err) {
-    try {
-      // Do no wrap rollovers in retryTransientEsErrors since it is not idempotent
-      const path = `/${dataStreamName}/_rollover`;
-      await esClient.transport.request({
-        method: 'POST',
-        path,
-      });
-    } catch (error) {
-      throw new Error(`cannot rollover data stream ${error}`);
-    }
+    await rolloverDataStream(dataStreamName, esClient);
+    return;
   }
   // update settings after mappings was successful to ensure
   // pointing to the new pipeline is safe
   // for now, only update the pipeline
+  const { settings } = indexTemplate.template;
```

Comment:
Does this have the fully resolved settings? It appears this comes from `getTemplate` which I don't think will have any settings: https://github.com/elastic/kibana/blob/acdac80d8e63b96178b3321fc137f37c10222f66/x-pack/plugins/fleet/server/services/epm/elasticsearch/template/template.ts/#L59

I think we should update with the fully resolved settings, maybe by fetching the new resolved template with the [template simulate API](https://www.elastic.co/guide/en/elasticsearch/reference/8.1/indices-simulate-template.html).

FWIW, I think this was a pre-existing issue with the current implementation, but as it is now in this PR it seems this will end up being a no-op now since I believe `settings.index.default_pipeline` will be undefined?

Reply:
This is a really good point, I'll look into it.